src/styles/main.scss

```diff
@@ -1,3 +1,94 @@
+* {
+  box-sizing: border-box;
+}
+
+html {
+  font-family: Arial, Helvetica, sans-serif;
+}
+
 body {
   margin: 0;
 }
+
+$day__color: #eee;
+$day__size: 95px;
+$col-number: 7;
+$col-padding: 10px;
+$gap: 1px;
+
+@mixin day($size) {
+  width: $size;
+  height: $size;
+}
+
+@function getWidth($size, $gap, $quantity, $padding) {
+  @return ($size + $gap) * $quantity + $padding;
+}
+
+.page__calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: getWidth($day__size, $gap, $col-number, $col-padding);
+
+  gap: 1px;
+  margin: 0 auto;
+  padding-top: 50px;
+}
+
+.calendar__day {
+  position: relative;
+
+  @include day($day__size);
+
+  background-color: $day__color;
+  border: 1px solid #000;
+  transition: background-color 0.5s, transform 0.5s;
+
+  &:hover {
+    cursor: pointer;
+    background-color: pink;
+    transform: translateY(-20px);
+  }
+}
+
+@for $num from 28 through 31 {
+  .calendar--month-length_#{$num} {
+    @for $day-num from $num + 1 through 32 {
+      .calendar__day:nth-child(#{$day-num}) {
+        display: none;
+      }
+    }
+  }
+}
+
+@for $num from 1 to 32 {
+  .calendar__day:nth-child(#{$num})::after {
+    content: "#{$num}";
+
+    position: absolute;
+    left: 50%;
+    top: 50%;
+
+    font-size: 30px;
+    color: #000;
+
+    transform: translate(-50%, -50%);
```

Comment:
add all these styles outside the loop, leave in the loop styles that depend only on the iteration of the loop